src/util/TxResource.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+// Copyright 2023 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "util/numeric.h"
+#include <algorithm>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace stellar
+{
+
+constexpr size_t NUM_CLASSIC_TX_RESOURCES(1);
+constexpr size_t NUM_SOROBAN_TX_RESOURCES(6);
+
+// Small helper class to allow arithmetic operations on tuples
+class Resource
```

Comment:
Agreed. This exact change was in my TODO list actually, but I didn't get to it because it was a bit annoying to propagate it to all the users of Resource (tx queue, limiter, etc). I do think it's worth doing to avoid any footguns. I'll submit a patch as a follow-up after we merge this change.

Reply:
Sounds good, we can do that as a followup after the preview release (if at all) - when I left this comment I didn't know the scope of usage of this class. Propagating it everywhere would likely be really awkward; maybe the runtime checks are good enough of a tradeoff here.